Battle.js

```diff
@@ -0,0 +1,124 @@
+/*TODO: 
+Main battle screen, should display 2 knights with health bars and a 
+button for each different ability*/
+
+import { StyleSheet, Text, View, Button, Image, Dimensions, Pressable } from 'react-native';
+import React, { useState, useRef, useEffect } from 'react';
+import { HealthBar } from "./HealthBar.js"
+import { score } from "./Dodge.js"
+import { useFocusEffect } from '@react-navigation/native';
+import { hookstate, useHookstate, createState } from '@hookstate/core';
```

Comment:
This is the "react" way of doing things, you can probably make it simpler than this, but good job for working on it yourself.

Video games are often very active, e.g. the react-native-game-engine class updates and re-renders every component 30 or 60 times a second. You could actually make the Battle Screen render as another <GameEngine> to allow for this kind of fast updating (or just because it is more familiar at this point).

The "react" way of doing things is designed for web-apps, which often contain a lot of components that only need to update occasionally, when something changes, or a user clicks on them. Rather than a loop that *always* triggers 30 frames a second, the model here is that most components aren't re-drawn (you can update variables, but without a re-render, the screen itself wont change) most of the time, and only "react" (hence the name) to certain events, causing a re-render to happen.

I would suggest starting with React's state, rather than using the Hookstate library, which does some fancy things (probably making it easier, but also less clear what depends on what). Take a look at this part of the tutorial: https://reactjs.org/tutorial/tutorial.html#making-an-interactive-component
Or a slightly more comprehensive overview here: https://reactjs.org/docs/state-and-lifecycle.html

Basically, react just gives you a special place to store "state" (i.e. variables at that update) and automatically re-draws the component when it's state changes:
```
class Battle {
  constructor() {
    
  }
}
```

TODO(braid): Finish example